pkg/workflow/providers/oam/apply.go

```diff
@@ -146,10 +206,82 @@ func lookUpValues(v *value.Value) (*common.ApplicationComponent, *value.Value, s
 	return comp, patcher, clusterName, overrideNamespace, env, nil
 }
 
+func (p *provider) loadDynamicComponent(comp *common.ApplicationComponent) (*common.ApplicationComponent, error) {
+	if comp.Type != "ref-objects" {
+		return comp, nil
+	}
+	_comp := comp.DeepCopy()
+	props := &struct {
+		Objects []struct {
+			APIVersion string            `json:"apiVersion"`
+			Kind       string            `json:"kind"`
+			Name       string            `json:"name,omitempty"`
+			Selector   map[string]string `json:"selector,omitempty"`
```

Comment:
obscure naming, is it field-selector or the label? either has built-in support in kubernetes